sklearn/tree/_criterion.pyx

```diff
@@ -1305,23 +1308,3 @@ cdef class FriedmanMSE(MSE):
                 self.weighted_n_left * total_sum_right)
 
         return diff * diff / (self.weighted_n_left * self.weighted_n_right)
-
-    cdef double impurity_improvement(self, double impurity) nogil:
```

Comment:
What is the reason for this change here?
Are we calling only the `proxy_impurity_improvement`?

Reply:
uhm it seems that with this change we will use the MSE impurity improvement

Reply:
Good catch, thanks. I didn't realize Friedman MSE used a non-conventional improvement. Which makes me going through a rabbit hole, wondering whether this makes sense at all, but that's irrelevant for this PR. I put it back

Reply:
Went down the same rabbit hole. :)

Reply:
how did you get out? I'm still super confused about so many things. Like does friedman_mse make sense outside of GBDTs, and do we really want to allow a MAE splitting criteria when we already have the LAD loss... so many questions lol

Reply:
I am not planning to get out anytime soon. There isn't many references of this criterion outside of https://statweb.stanford.edu/~jhf/ftp/trebst.pdf

Reply:
Yeah... Here are my notes so far. I'll open an issue when I have a better idea of this all, but I'm happy to sync with you prior!


Does it really make sense to allow a criterion to be passed to GBDT? All trees
should be built via LS anyway. Friedman does mention e.g. for LAD that trees
could be built with LAD criterion but LS is just much faster.

WTF is friedman_mse?
- was introduced here https://github.com/scikit-learn/scikit-learn/pull/2570
- it used to be the hardcoded and non-overidable default for all GBDT
- Then https://github.com/scikit-learn/scikit-learn/pull/6667 introduced LAD
  criterion for all trees, and started exposing a criterion param to all
  models.
- Does it really make sense to allow it for all tree models??
- It was introduced in the context of **multiclass GB**.
- Also WTF are the weights? Our implementation completely differs from what the 
  paper defines. the weights in the paper aren't sample weights, they're the hessians

Reply:
🥕 ?
@NicolasHug I was thinking/doubting about friedman_mse myself and would appreciate an issue if you come up with one.